src/relay/op/tensor/transform.cc

```diff
@@ -3146,5 +3146,84 @@ RELAY_REGISTER_OP("matrix_set_diag")
     .set_attr<FTVMCompute>("FTVMCompute", MatrixSetDiagCompute)
     .set_attr<TOpPattern>("TOpPattern", kInjective);
 
+// adv_index
+bool AdvIndexRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
+                 const TypeReporter& reporter) {
+  CHECK_EQ(num_inputs, 1);
+  auto inputs = types[0].as<TupleTypeNode>();
+  CHECK(inputs != nullptr);
```

Comment:
Don't CHECK here, it will fail in partial type inference (ran into a lot of issues like this in the dynamic ONNX PR). Instead, do

```
if (inputs == nullptr) {
    return false;
}
```

Reply:
@mbrookhart could you explain a little more?

Reply:
I'd leave it to @jroesch to explain the InferType algorithm more fully, but the basic issue is that sometimes the InferType algorithm will call this function with an as-of yet incomplete input type, I think the class is `IncompleteTypeNode`. If we CHECK here, that crashes the algorithm, but if we return false (i.e., we failed to infer the output type), the algorithm will try again once it has more information on the inputs, at which point we will pass the check and continue with the function.